.make.defaults

```diff
@@ -51,7 +51,7 @@ DOCKER_LOCAL_IMAGE=$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_VERSION)
 DOCKER_REMOTE_IMAGE=$(DOCKER_REGISTRY_ENDPOINT)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_VERSION)
 DOCKER_SPARK_BASE_IMAGE_NAME=data-prep-kit-spark-3.5.2
 DOCKER_SPARK_BASE_IMAGE=$(DOCKER_SPARK_BASE_IMAGE_NAME):$(DOCKER_IMAGE_VERSION)
-RAY_BASE_IMAGE?=docker.io/rayproject/ray:${RAY}-py310
+RAY_BASE_IMAGE?=docker.io/rayproject/ray:${RAY}-py312
```

Comment:
Can we make Python version a separate parameter in make.versions? I am sure it can be used elsewhere

Reply:
Also this is kinda dangerous. what if I am on Python3.10? It really should get current Python version and adjust the base Ray image accordingly

Reply:
No, here the question is the Python version of the images we deploy to the repository. This includes the RayKFP image. It doesn't connect to the Python version of a certain user. 
The original issue is to support Python v3.12, now we do support it. Should we change our Python default version to v3.12? probably not.

I'm going to return it to v3.10, but I'm open to other ideas. 
 
 CC: @daw3rd , @shahrokhDaijavad @yuanchi2807 @Bytes-Explorer @blublinsky

Reply:
Aren't images different than venv?  Base images can be whatever works, as long as we do the transform tests in the image, which we are.  

Separately, we probably need to start using a github workflow matrix to test at least 3.11 and 3.12.  This requires changing .github/workflow/test-transform.template somehow and then regenerating them with `make`.

Reply:
Ray is very sensitive to its and Python versions; they should be the same on clients and servers. We should provide a set of images, or users will have to build the images by themselves.

Reply:
OK, I validated that all tests passed with Python 3.12, not I will return the default setting to 3.10.